--- src/Makefile.global.in.orig	2014-01-20 15:02:43.997373163 -0800
+++ src/Makefile.global.in	2014-01-20 15:07:44.810346342 -0800
@@ -60,7 +60,7 @@
 #
 # These are set by the equivalent --xxxdir configure options.  We
 # append "postgresql" to some of them, if the string does not already
-# contain "pgsql" or "postgres", in order to avoid directory clutter.
+# contain "pgxc", "pgsql", or "postgres", in order to avoid directory clutter.
 #
 # In a PGXS build, we cannot use the values inserted into Makefile.global
 # by configure, since the installation tree may have been relocated.
@@ -78,43 +78,53 @@
 bindir := @bindir@
 
 datadir := @datadir@
+ifeq "$(findstring pgxc, $(datadir))" ""
 ifeq "$(findstring pgsql, $(datadir))" ""
 ifeq "$(findstring postgres, $(datadir))" ""
-override datadir := $(datadir)/postgresql
+override datadir := $(datadir)/pgxc
+endif
 endif
 endif
 
 sysconfdir := @sysconfdir@
+ifeq "$(findstring pgxc, $(datadir))" ""
 ifeq "$(findstring pgsql, $(sysconfdir))" ""
 ifeq "$(findstring postgres, $(sysconfdir))" ""
-override sysconfdir := $(sysconfdir)/postgresql
+override sysconfdir := $(sysconfdir)/pgxc
+endif
 endif
 endif
 
 libdir := @libdir@
 
 pkglibdir = $(libdir)
+ifeq "$(findstring pgxc, $(datadir))" ""
 ifeq "$(findstring pgsql, $(pkglibdir))" ""
 ifeq "$(findstring postgres, $(pkglibdir))" ""
-override pkglibdir := $(pkglibdir)/postgresql
+override pkglibdir := $(pkglibdir)/pgxc
+endif
 endif
 endif
 
 includedir := @includedir@
 
 pkgincludedir = $(includedir)
+ifeq "$(findstring pgxc, $(datadir))" ""
 ifeq "$(findstring pgsql, $(pkgincludedir))" ""
 ifeq "$(findstring postgres, $(pkgincludedir))" ""
-override pkgincludedir := $(pkgincludedir)/postgresql
+override pkgincludedir := $(pkgincludedir)/pgxc
+endif
 endif
 endif
 
 mandir := @mandir@
 
 docdir := @docdir@
+ifeq "$(findstring pgxc, $(datadir))" ""
 ifeq "$(findstring pgsql, $(docdir))" ""
 ifeq "$(findstring postgres, $(docdir))" ""
-override docdir := $(docdir)/postgresql
+override docdir := $(docdir)/pgxc
+endif
 endif
 endif
 
--- doc/src/sgml/installation.sgml.orig	2014-01-20 15:11:22.673376445 -0800
+++ doc/src/sgml/installation.sgml	2014-01-20 15:10:55.199346105 -0800
@@ -567,7 +567,7 @@
          Sets the root directory for installing documentation files,
          except <quote>man</> pages.  This only sets the default for
          the following options.  The default value for this option is
-         <filename><replaceable>DATAROOTDIR</>/doc/postgresql</>.
+         <filename><replaceable>DATAROOTDIR</>/doc/pgxc</>.
         </para>
        </listitem>
       </varlistentry>
@@ -591,14 +591,14 @@
        <productname>PostgreSQL</> into shared installation locations
        (such as <filename>/usr/local/include</filename>) without
        interfering with the namespace of the rest of the system. First,
-       the string <quote><literal>/postgresql</literal></quote> is
+       the string <quote><literal>/pgxc</literal></quote> is
        automatically appended to <varname>datadir</varname>,
        <varname>sysconfdir</varname>, and <varname>docdir</varname>,
        unless the fully expanded directory name already contains the
-       string <quote><literal>postgres</></quote> or
+       string <quote><literal>postgres</></quote>, <quote><literal>pgxc</></quote>, or
        <quote><literal>pgsql</></quote>. For example, if you choose
        <filename>/usr/local</filename> as prefix, the documentation will
-       be installed in <filename>/usr/local/doc/postgresql</filename>,
+       be installed in <filename>/usr/local/doc/pgxc</filename>,
        but if the prefix is <filename>/opt/postgres</filename>, then it
        will be in <filename>/opt/postgres/doc</filename>. The public C
        header files of the client interfaces are installed into
